src/components/Dots.js

```diff
@@ -0,0 +1,24 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Dot from './Dot';
+import './Dots.css';
+
+class Dots extends PureComponent {
+  render() {
+    const { index, quantity, dotClick } = this.props;
+    let dots = [];
+    for (let i = 0; i < quantity; i++) {
+      dots.push(<Dot key={i} id={i} active={i === index} dotClick={dotClick} />);
+    }
+
+    return <div className="wui-dots-container">{dots}</div>;
+  }
+}
+
+Dot.propTypes = {
+  index: PropTypes.number,
+  quantity: PropTypes.number,
+  dotClick: PropTypes.func,
```

Comment:
func.isRequired

if not required you need to set default props

Reply:
If I make a default function, my component lost sense. If I set it as required, the console throw some warnings.